api/main.go

```diff
@@ -1,13 +1,66 @@
 package main
 
-import "os"
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/joho/godotenv"
+)
 
 func main() {
-	a := App{}
-	a.Initialize(
-		os.Getenv("POSTGRES_USER"),
+	flagUseEnv := flag.String("env", "", "Provides the path to .env files")
+
+	flag.Parse()
+
+	if *flagUseEnv != "" {
+		fmt.Printf("Loading .env : %v ", *flagUseEnv)
+		err := godotenv.Load(*flagUseEnv)
+		if err != nil {
+			log.Fatal("Error loading .env file")
+		}
+		fmt.Println("✅")
+	}
+
+	checkEnv()
+
+	initDB(os.Getenv("POSTGRES_USER"),
 		os.Getenv("POSTGRES_PASSWORD"),
 		os.Getenv("POSTGRES_DB"),
-		os.Getenv("POSTGRES_HOST"))
+		os.Getenv("POSTGRES_HOST"),
+		os.Getenv("POSTGRES_PORT"))
+
+	a := App{}
+	a.Initialize()
 	a.Run(":" + os.Getenv("API_PORT"))
 }
+
+// initDB creates a global connection pool from identifiers.
+// DB is global, because it is the connection pool not the connection itself.
```

Comment:
Wouldn't it be better to pass to the Initialize method instead of using a global variable ?